contracts/utils/Swap.sol

```diff
@@ -162,17 +157,29 @@ contract Swap is ISwap, ReentrancyGuard, Ownable {
     {
         Offer storage offer = swapOffers[swapId];
 
-        require(offer.seller != address(0), "Offer does not exist");
-        require(msg.sender == offer.seller, "Only seller can settle");
-        require(offer.isOpen, "Offer already closed");
+        address seller = offer.seller;
+        require(seller != address(0), "Offer does not exist");
+        require(seller == msg.sender, "Only seller can settle");
```

Comment:
`require(seller != address(0))` and `require(seller == msg.sender)` are redundant assertions because if seller == msg.sender, then it implies seller != address(0).

To save gas, we can simply do:

```solidity
require(seller == msg.sender, "Only seller can settle or offer doesn't exist");
```